### DISCUSSION TITLE
Customs prompts always include default system message

### DISCUSSION THREAD

**Discussion by Holmgren825** [olimorris/codecompanion.nvim#497] [Q&A] [answered]
Hey!
Not sure if this is a bug or a feature, hence the discussion. I've tried to set up a custom prompt based on the recipe "My New Prompt", and changing the content of the system role. But when I start a chat with the new action, the default system prompt "You are an AI programming assistant named \"CodeCompanion\"...." is still included, followed by the new system prompt (as can be seen in the debug info.

 I figured that the custom prompt should replace the default one. But maybe it is just my understanding that is wrong?

**Comment by olimorris** [comment, marked as answer]
That's a very good point.

The system prompt is sent by default with all requests (you can remove this in the chat buffer with `gs` or setting its config value to `""`). But there should be an option in your custom prompts to _not_ send the default system prompt. I'll put this down as an enhancement and try to get to it early this week.

Thanks for raising.

**Reply by Holmgren825** [reply]
Sounds good! Thanks for the quick reply, and your work!

**Reply by olimorris** [reply]
This is now in `v10.4.0`. So in your custom prompts, set `ignore_system_prompt = true`:

```lua
["Your_New_Prompt"] = {
  strategy = "chat",
  description = "Your Special New Prompt",
  opts = {
    ignore_system_prompt = true,
  },
  -- Your prompts here
}
```
